Skip 2 unit tests if bio.tools is down

nsoranzo opened the pull request:
Also:
- Set default [verbosity](https://docs.pytest.org/en/7.1.x/how-to/output.html#verbosity) of pytest to 1. This is equivalent to the `-v` option, which e.g. shows each test inside a file as a separate line in the output, showing the reason why a test was skipped.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

nsoranzo committed:
Set default verbosity of pytest to 1

This is equivalent to the `-v` option, which e.g. shows each test inside a
file as a separate line in the output, showing the reason why a test was
skipped.

xref. https://docs.pytest.org/en/7.1.x/how-to/output.html#verbosity

nsoranzo committed:
Make 2 test decorators usable on both functions and methods

nsoranzo committed:
Skip unit tests requiring bio.tools if it's down

jdavcs reviewed (approved):
The bio.tools tests are no longer down, but I see no harm in adding the decorator.
And I do prefer pytest's verbose output - thanks!

nsoranzo committed:
Move ``skip_if_site_down`` test decorator to be reusable in unit tests

Also:
- Add type annotations

nsoranzo committed:
Add missing package requirements

Packages that are imported directly in galaxy-tool-util that were already
required through the galaxy-util dependency.
